integration/load.php

```diff
@@ -162,3 +162,12 @@ function ( $post_types ) {
  * @see https://buddypress.org/
  */
 add_action( 'bp_include', array( __NAMESPACE__ . '\Buddypress', 'init' ), 0 );
+
+/**
+ * Load the Surge integration.
+ *
+ * Only load code that needs Surge to run once Surge is loaded and initialized.
+ *
+ * @see https://wordpress.org/plugins/surge/
+ */
+add_action( 'init', array( __NAMESPACE__ . '\Surge', 'init' ) );
```

Comment:
Adding this to `init` runs it after the "regular" integration inits, which run on `plugins_loaded`. I think it's fine to move this into `plugin_init()` and just call it directly `Surge::init()`.

That said, to account for situations where Surge is already active, I think we should add activation/deactivation callbacks for activitypub here:
```
\add_action( 'activate_activitypub/activitypub.php', array( __NAMESPACE__ . '\Surge', 'add_cache_config' ) );
\add_action( 'deactivate_activitypub/activitypub.php', array( __NAMESPACE__ . '\Surge', 'remove_cache_config' ) );
```

Reply:
can a plugin listen to it's own activation/deactivation? I think we would have to use the `register_activation_hook` instead?

this might get complicated and I think we should deal with that in a follow up!?!

it still will be caught with the site-health check.

Reply:
This is all `register_activation_hook` does, adding that action. It just makes it easier by not requiring plugin authors to know their plugin's hook. But it's a good point, we could probably just use that here:

```
\register_activation_hook( ACTIVITYPUB_PLUGIN_FILE, array( __NAMESPACE__ . '\Surge', 'add_cache_config' ) );
\register_deactivation_hook( ACTIVITYPUB_PLUGIN_FILE, array( __NAMESPACE__ . '\Surge', 'remove_cache_config' ) );
```

> this might get complicated and I think we should deal with that in a follow up!?!

I tested it (with the previous hooks), it should be as easy as adding those two lines.

Reply:
where have you added these lines? in `load.php`?

Reply:
Yeah, right below this init callback.